ext/DeviceUtilsCUDAExt.jl

```diff
@@ -0,0 +1,133 @@
+module DeviceUtilsCUDAExt
+
+using Adapt: Adapt
+using CUDA: CUDA
+using CUDA.CUSPARSE: AbstractCuSparseMatrix, AbstractCuSparseVector
+using DeviceUtils: DeviceUtils, CUDADevice, CPUDevice, reset_gpu_device!
+using Random: Random
+using cuDNN: cuDNN
+
+__init__() = reset_gpu_device!()
+
+const USE_CUDA_GPU = Ref{Union{Nothing, Bool}}(nothing)
+
+function _check_use_cuda!()
+    USE_CUDA_GPU[] === nothing || return
```

Comment:
Move this into `CUDAcuDNN` Ext similar to the `LuxCUDA` before. `CUDADevice` should work if CUDA.jl is loaded. The only thing `cuDNN` ensures is that `CUDADevice` becomes part of the automatic device selection. For example:


```julia
using DeviceUtils, CUDA

gdev = gpu_device() # Returns CPU Device

gdev_manual = CUDADevice() # Note that this works!

using cuDNN

gdev = gpu_device() # Now this returns CUDADevice
```